File: library/src/main/java/com/goyourfly/tabviewpager/IsViewPagerAnimating.java
package com.goyourfly.tabviewpager;

import android.support.v4.view.ViewPager;
import android.util.Log;
import android.widget.Scroller;

import java.lang.reflect.Field;

/**
 * Created by gaoyufei on 2017/12/2.
 */

public class IsViewPagerAnimating {
    public static final boolean isAnimating(ViewPager viewPager){
        Class klass = ViewPager.class;
        Scroller scroller = null;
        try {
            Field field = klass.getDeclaredField("mScroller");
            field.setAccessible(true);
            scroller = (Scroller) field.get(viewPager);
            return !scroller.isFinished();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        } catch (NoSuchFieldException e) {
            e.printStackTrace();
        }
        return false;
    }
}

File: library/src/main/java/com/goyourfly/tabviewpager/IsViewPagerAnimating.kt
package com.goyourfly.tabviewpager

import android.support.v4.view.ViewPager
import android.widget.Scroller

/**
 * Created by gaoyufei on 2017/12/2.
 * 这个类用来判断ViewPager是否正在做动画
 */

object IsViewPagerAnimating {
    fun isAnimating(viewPager: ViewPager): Boolean {
        val klass = ViewPager::class.java
        var scroller: Scroller? = null
        try {
            val field = klass.getDeclaredField("mScroller")
            field.isAccessible = true
            scroller = field.get(viewPager) as Scroller
            return !scroller.isFinished
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        } catch (e: NoSuchFieldException) {
            e.printStackTrace()
        }
        return false
    }
}
